
float64 xd
float64 yd
float64 phid
